File: tests/leetcode/x/dart/0060.dart
import 'dart:convert';
import 'dart:io';

String getPermutation(int n, int kInput) {
  final digits = List<String>.generate(n, (i) => '${i + 1}');
  final fact = List<int>.filled(n + 1, 1);
  for (var i = 1; i <= n; i++) fact[i] = fact[i - 1] * i;
  var k = kInput - 1;
  final out = StringBuffer();
  for (var rem = n; rem >= 1; rem--) {
    final block = fact[rem - 1];
    final idx = k ~/ block;
    k %= block;
    out.write(digits.removeAt(idx));
  }
  return out.toString();
}

Future<void> main() async {
  final lines = const LineSplitter().convert(await stdin.transform(utf8.decoder).join());
  if (lines.isEmpty || lines[0].trim().isEmpty) return;
  var idx = 0;
  final t = int.parse(lines[idx++].trim());
  final out = <String>[];
  for (var tc = 0; tc < t; tc++) {
    final n = int.parse(lines[idx++].trim());
    final k = int.parse(lines[idx++].trim());
    out.add(getPermutation(n, k));
  }
  stdout.write(out.join('\n'));
}

File: tests/leetcode/x/csharp/0060.cs
using System;
using System.Collections.Generic;

class Program {
    static string GetPermutation(int n, int k) {
        var digits = new List<string>();
        for (int i = 1; i <= n; i++) digits.Add(i.ToString());
        var fact = new int[n + 1];
        fact[0] = 1;
        for (int i = 1; i <= n; i++) fact[i] = fact[i - 1] * i;
        k--;
        var outp = "";
        for (int rem = n; rem >= 1; rem--) {
            int block = fact[rem - 1];
            int idx = k / block;
            k %= block;
            outp += digits[idx];
            digits.RemoveAt(idx);
        }
        return outp;
    }

    static void Main() {
        var lines = Console.In.ReadToEnd().Split(new[] { "\r\n", "\n" }, StringSplitOptions.None);
        if (lines.Length == 0 || lines[0].Trim() == "") return;
        int idx = 0, t = int.Parse(lines[idx++].Trim());
        var outp = new List<string>();
        for (int tc = 0; tc < t; tc++) {
            int n = int.Parse(lines[idx++].Trim());
            int k = int.Parse(lines[idx++].Trim());
            outp.Add(GetPermutation(n, k));
        }
        Console.Write(string.Join("\n", outp));
    }
}
